torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +306,27 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    # Sometimes by the time we get to tensorify, there have already been
+    # specializations, eg. in python_arg_parser.h. In these cases,
+    # placeholder nodes no longer have a reference to their original
+    # symfloat and thus we need to deduce specializations have happend
+    # via shape_env.replacements. NB: there's an important invariant here
+    # that symfloats keep consistent names across restarts.
```

Comment:
So what, you have to allocate the symfloat name even if you're not going to use it at all? Seems sus lol

Reply:
Don't we associate every var with sources?

```
        self.var_to_sources: Dict[sympy.Symbol, List[Source]] = {}
```

So that's how you get sources.